### ISSUE TITLE
Home button animation

### ISSUE THREAD

**Issue by oniony** [8bitPit/Niagara-Issues#1804]
Since I received version 1.5.2, the animation when returning to the homescreen from an app is broken.

Previous behaviour: hitting the home button would slide my current app downwards, revealing the homescreen and the icons would gently slide down into place across a few frames.

Current behviour: hitting the home button slides the app downwards revealing the launcher with the icons in their final positions and then clones of the icons slide down on top of them.

However, if I go into the app selector by hitting the  square icon then the correct animation (without duplicates) is shown. It's only when I switch into an app and return home do the duplicate icons animate into place.

I'm on a Nokia 8 running Android 9.

**Comment by oniony** [comment]
https://youtu.be/KJXdhy0kj8Q

**Comment by oniony** [comment]
I've just found that if I switch to a static wallpaper the problem goes away, so this appears to be an issue with live wallpapers.

**Comment by oniony** [comment]
Actually it happens with static wallpapers too, just not as predictably. With the static wallpaper it may happen some few seconds after returning to the launcher.

**Comment by Maxr1998** [comment]
I've seen this occur myself a few times, I suspect it's a bug in one of the libraries we use. Usually a launcher restart fixes it again, and it should only be triggered by updating or rebooting the device.

Does it happen more regularly for you?

**Comment by topsy-turvy99** [comment]
Me too, i face the same problem with live wallpaper, so i stopped using it, in static wallpaper i face the same sometimes, but i see the issue more when unlocking the phone, it's not smooth

**Comment by oniony** [comment]
> I've seen this occur myself a few times, I suspect it's a bug in one of the libraries we use. Usually a launcher restart fixes it again, and it should only be triggered by updating or rebooting the device.
> 
> Does it happen more regularly for you?

Yes, it happens all the time. Immediately after restarting the launcher (without pressing anything more) and following phone restart. In fact the only time it doesn't happen is when I go into the app switcher but don't actually switch.

**Comment by oniony** [comment]
I'm pretty sure it didn't happen before the update. If you have any APKs of older builds I'd be willing to install to confirm exact version.

**Comment by Maxr1998** [comment]
All APKs from older updates are attached to the releases of this repo.

**Comment by oniony** [comment]
OK, I've installed each version. The problem is evident with every 1.5.x version. When I went back to 1.4.2 the problem is completely gone.

**Comment by oniony** [comment]
Actually 1.4.2 is so silky smooth I think I'm just going to stick with that for now.

**Comment by mazikowski** [comment]
I'm having the same problem. It's almost as if there are two copies of the launcher stacked, one with the Niagara widget loaded and the other where it loads as it displays.

https://user-images.githubusercontent.com/8191823/148151787-51f55101-f7d2-497d-8c06-5d1296da53a3.mp4

**Comment by Maxr1998** [comment]
Both of the underlying issues were identified and fixed. Should be rolled out in the next release.

**Comment by oniony** [comment]
Awesome, thank you!

**Comment by hexboxlabs** [comment]
The same issue but while unlocking the phone still exist on V.1.6.5. (Galaxy A33)
Pressing home button, or changing apps from home screen seems fine.

**Comment by Maxr1998** [comment]
It indeed seems like this issue still pops up occasionally. I have seen at least one other report on a Samsung phone as well, and it once happened to me too.
I can't say if the fix didn't work in the first place or if it is a regression. I'll have to investigate.

**Comment by Oddward** [comment]
It seems like on my Galaxy A52 the behaviour is like this: 

- when I unlock the phone using the **on-screen fingerprint scanner** it has the slide down animation which is buggy and layers on top of an already static Niagara UI 
- when I unlock using the pattern (I assume it's the same with pin) the unlock animation is like a zoom out into place and it's smooth

The bug might be related to a specific unlock animation or something, at least on Samsung

**Comment by Amorphous404** [comment]
@oniony do you still happen to face the issue with the home button?

**Comment by Amorphous404** [comment]
> It indeed seems like this issue still pops up occasionally. I have seen at least one other report on a Samsung phone as well, and it once happened to me too.
> I can't say if the fix didn't work in the first place or if it is a regression. I'll have to investigate.

The latter issue did get fixed with the following updates on my older phone where i faced the issue with live wallpaper apps, but this seem to only get triggered via fingerprint unlock on some devices.

Not sure if i should merge this ticket https://github.com/8bitPit/Niagara-Issues/issues/2092 with this one 😅
